test_util/launch.py

```diff
@@ -61,40 +68,103 @@ def delete(self, info):
     def ssh_from_config(self, info):
         raise NotImplementedError()
 
-    def test(self, info):
-        raise NotImplementedError()
+    def test(self, info, test_cmd):
+        ssh_info = info['ssh']
+        try:
+            check_keys(ssh_info, ['user', 'private_key'])
+        except LauncherError:
+            print('DC/OS Launch is missing sufficient SSH info to run tests!')
+            raise
+        details = self.describe(info)
+        test_host = details['masters'][0]['public_ip']
+        with ssh.tunnel.tunnel(ssh_info['user'], ssh_info['private_key'], test_host) as test_tunnel:
+            return test_util.runner.integration_test(
+                tunnel=test_tunnel,
+                dcos_dns=test_host,
+                master_list=[m['private_ip'] for m in details['masters']],
+                agent_list=[a['private_ip'] for a in details['private_agents']],
+                public_agent_list=[a['private_ip'] for a in details['public_agents']],
+                aws_access_key_id=info['provider'].get('access_key_id'),
+                aws_secret_access_key=info['provider'].get('secret_access_key'),
+                region=info['provider'].get('region'),
+                test_cmd=test_cmd)
 
 
 class AwsCloudformationLauncher(AbstractLauncher):
     def __init__(self, boto_wrapper):
         self.boto_wrapper = boto_wrapper
 
     def create(self, config):
+        """
+        Checks config and creates missing resources as necessary
+        If actual stack creation fails, then network resources
+        will be cleaned up
+
+        Note: ssh_from_config and provide_network_prereqs will mutate
+            config
+        """
         check_keys(config, ['stack_name', 'template_url'])
-        ssh_info = self.ssh_from_config(config)
-        # NOTE: even if parameters not given, ssh_from_config will add KeyName
-        # parameter to config['parameters']
-        self.boto_wrapper.create_stack(
-            config['stack_name'], config['template_url'], config['parameters'])
+        ssh_info, temp_resources = self.ssh_from_config(config)
+        if config.get('zen_helper', False):
+            temp_resources.update(self.provide_network_prereqs(config))
+        try:
+            self.boto_wrapper.create_stack(
+                config['stack_name'], config['template_url'], config['parameters'])
+        except Exception as ex:
```

Comment:
No, any exception. If the launch fails for *any* reason, we want to delete all the resources that were implicitly created if we can